api/plugin_helpers.go

```diff
@@ -23,6 +24,35 @@ var (
 	// PluginUnwrapTokenEnv is the ENV name used to pass unwrap tokens to the
 	// plugin.
 	PluginUnwrapTokenEnv = "VAULT_UNWRAP_TOKEN"
+
+	sudoPaths = map[string]*regexp.Regexp{
+		"/auth/token/accessors/":                        regexp.MustCompile("^/auth/token/accessors/$"),
+		"/pki/root":                                     regexp.MustCompile("^/pki/root$"),
+		"/pki/root/sign-self-issued":                    regexp.MustCompile("^/pki/root/sign-self-issued$"),
+		"/sys/audit":                                    regexp.MustCompile("^/sys/audit$"),
+		"/sys/audit/{path}":                             regexp.MustCompile("^/sys/audit/.+$"),
+		"/sys/auth/{path}":                              regexp.MustCompile("^/sys/auth/.+$"),
+		"/sys/auth/{path}/tune":                         regexp.MustCompile("^/sys/auth/.+/tune$"),
+		"/sys/config/auditing/request-headers":          regexp.MustCompile("^/sys/config/auditing/request-headers$"),
+		"/sys/config/auditing/request-headers/{header}": regexp.MustCompile("^/sys/config/auditing/request-headers/.+$"),
+		"/sys/config/cors":                              regexp.MustCompile("^/sys/config/cors$"),
+		"/sys/config/ui/headers/":                       regexp.MustCompile("^/sys/config/ui/headers/$"),
+		"/sys/config/ui/headers/{header}":               regexp.MustCompile("^/sys/config/ui/headers/.+$"),
+		"/sys/leases":                                   regexp.MustCompile("^/sys/leases$"),
+		"/sys/leases/lookup/":                           regexp.MustCompile("^/sys/leases/lookup/$"),
+		"/sys/leases/lookup/{prefix}":                   regexp.MustCompile("^/sys/leases/lookup/.+$"),
+		"/sys/leases/revoke-force/{prefix}":             regexp.MustCompile("^/sys/leases/revoke-force/.+$"),
+		"/sys/leases/revoke-prefix/{prefix}":            regexp.MustCompile("^/sys/leases/revoke-prefix/.+$"),
+		"/sys/plugins/catalog/{name}":                   regexp.MustCompile("^/sys/plugins/catalog/.+$"),
+		"/sys/plugins/catalog/{type}":                   regexp.MustCompile("^/sys/plugins/catalog/.+$"),
+		"/sys/plugins/catalog/{type}/{name}":            regexp.MustCompile("^/sys/plugins/catalog/.+/.+$"),
```

Comment:
Should we limit then non-`{path}` regexes to something like `[\w-_]+` instead of `.+` or would it be too restrictive?